src/omv/modules/py_image.c

```diff
@@ -5096,190 +5096,123 @@ static void py_apriltag_print(const mp_print_t *print, mp_obj_t self_in, mp_prin
               (double) mp_obj_get_float(self->z_rotation));
 }
 
-static mp_obj_t py_apriltag_subscr(mp_obj_t self_in, mp_obj_t index, mp_obj_t value) {
-    if (value == MP_OBJ_SENTINEL) {
-        // load
-        py_apriltag_obj_t *self = self_in;
-        if (MP_OBJ_IS_TYPE(index, &mp_type_slice)) {
-            mp_bound_slice_t slice;
-            if (!mp_seq_get_fast_slice_indexes(py_apriltag_obj_size, index, &slice)) {
-                mp_raise_msg(&mp_type_OSError, MP_ERROR_TEXT("only slices with step=1 (aka None) are supported"));
-            }
-            mp_obj_tuple_t *result = mp_obj_new_tuple(slice.stop - slice.start, NULL);
-            mp_seq_copy(result->items, &(self->x) + slice.start, result->len, mp_obj_t);
-            return result;
-        }
-        switch (mp_get_index(self->base.type, py_apriltag_obj_size, index, false)) {
-            case 0: return self->x;
-            case 1: return self->y;
-            case 2: return self->w;
-            case 3: return self->h;
-            case 4: return self->id;
-            case 5: return self->family;
-            case 6: return self->cx;
-            case 7: return self->cy;
-            case 8: return self->rotation;
-            case 9: return self->decision_margin;
-            case 10: return self->hamming;
-            case 11: return self->goodness;
-            case 12: return self->x_translation;
-            case 13: return self->y_translation;
-            case 14: return self->z_translation;
-            case 15: return self->x_rotation;
-            case 16: return self->y_rotation;
-            case 17: return self->z_rotation;
+static void py_apriltag_attr(mp_obj_t self_in, qstr attr, mp_obj_t *dest) {
+    py_apriltag_obj_t *self = MP_OBJ_TO_PTR(self_in);
+    if (dest[0] == MP_OBJ_NULL) {
+        // Load attribute.
+        switch (attr) {
+            case MP_QSTR_corners:
+                dest[0] = self->corners;
+                break;
+            case MP_QSTR_rect:
+                dest[0] = mp_obj_new_tuple(4, (mp_obj_t []) { self->x, self->y, self->w, self->h });
+                break;
+            case MP_QSTR_x:
+                dest[0] = self->x;
+                break;
+            case MP_QSTR_y:
+                dest[0] = self->y;
+                break;
+            case MP_QSTR_w:
+                dest[0] = self->w;
+                break;
+            case MP_QSTR_h:
+                dest[0] = self->h;
+                break;
+            case MP_QSTR_id:
+                dest[0] = self->id;
+                break;
+            case MP_QSTR_family:
+                dest[0] = self->family;
+                break;
+            case MP_QSTR_family_str:
+                #ifdef IMLIB_ENABLE_APRILTAGS_TAG16H5
```

Comment:
There's no need for this `apriltag_families` are always defined, a switch statement here is fine even if not all of them are enabled.

Reply:
Done